source/GameInterface/Services/MobileParties/Patches/PlayerLeaveSettlementPatch.cs

```diff
@@ -34,6 +34,8 @@ public static void OverrideLeaveConsequence()
             LeaveSettlementActionPatches.AllowedInstance.Instance = MobileParty.MainParty;
             PlayerEncounter.Finish(true);
             Campaign.Current.SaveHandler.SignalAutoSave();
+
+            EncounterManagerPatches.inSettlement = false;
```

Comment:
Move this into a patch for the PlayerEncounter.Finish method in the PlayerEncounterPatches.cs file